zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterProcess.java

```diff
@@ -140,7 +140,27 @@ public Client getClient() throws Exception {
   }
 
   public void releaseClient(Client client) {
-    clientPool.returnObject(client);
+    releaseClient(client, false);
+  }
+
+  public void releaseClient(Client client, boolean broken) {
+    if (broken) {
+      releaseBrokenClient(client);
+    } else {
+      clientPool.returnObject(client);
+    }
+  }
+
+  public void releaseBrokenClient(Client client) {
+    try {
+      clientPool.invalidateObject(client);
+    } catch (Exception e) {
+      if (e instanceof RuntimeException) {
+        throw (RuntimeException) e;
```

Comment:
Actually releaseClient() also has this issue since clientPool.returnObject() can throw similar kind of runtime exceptions. 
Just difference between `returnObject()` and `invalidateObject()` is invalidateObject() throws checked exception.

If we want to release client quietly (consume any exceptions), I can address to releaseClient() and releaseBrokenClient().